inc/item_operatingsystem.class.php

```diff
@@ -0,0 +1,379 @@
+<?php
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access directly to this file");
+}
+
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2017 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+**/
+class Item_OperatingSystem extends CommonDBRelation {
+
+   static public $itemtype_1 = 'OperatingSystem';
+   static public $items_id_1 = 'operatingsystems_id';
+   static public $itemtype_2 = 'itemtype';
+   static public $items_id_2 = 'items_id';
+   static public $checkItem_2_Rights = self::DONT_CHECK_ITEM_RIGHTS;
```

Comment:
I'm not sure about this right flag.
I think revert to the other side the right could be better and make more sense. aka:
``` static public $checkItem_1_Rights = self::DONT_CHECK_ITEM_RIGHTS;```

I think, when you add an os to a computer, you want to check if you have right on the computer, not the right on the os.